lang/cpp.go

```diff
@@ -0,0 +1,470 @@
+package lang
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+
+	"github.com/j178/leetgo/leetcode"
+)
+
+type cpp struct {
+	baseLang
+}
+
+//go:embed cpp/LC_DS.h
+var structDefi string
+
+//go:embed cpp/LC_IO.h
+var helperFuncs string
+
+var cppTypes = map[string]string{
+	"void":     "void",
+	"integer":  "int",
+	"long":     "int64_t",
+	"string":   "string",
+	"double":   "double",
+	"ListNode": "ListNode*",
+	"TreeNode": "TreeNode*",
+}
+
+const (
+	objectName            = "obj"
+	returnName            = "res"
+	inputFileStreamName   = "ifs"
+	outputFileStreamName  = "ofs"
+	systemDesignFuncName  = "sys_design_func"
+	systemDesignFuncNames = "sys_design_funcs"
+	cppTestFileTemplate   = `#include <bits/stdc++.h>
+using namespace std;
+
+// definitions
+%s
+
+#include "solution.h"
+
+// helper funcs
+%s
+
+// main func
+int main(int argc, char **argv) {
+	if (argc != 3) {
+		return 1;
+	}
+
+	ifstream ` + inputFileStreamName + `(argv[1]);
```

Comment:
How about just hard-coded `ifs` here and remove the variable?

Reply:
Could be okay, but I was worried that this name can be in conflict with the names from some of the problems, so I made it like this which can be changed globally if it become an issue in the future.